File: Data Structures & Algorithms/subsets/submission-1.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        curr = []
        self.backtrack(0, curr, res, nums)
        return res

    def backtrack(self, start, curr, res, nums):
        res.append(curr.copy())
        for i in range(start, len(nums)):
            curr.append(nums[i])
            self.backtrack(i + 1, curr, res, nums)
            curr.pop()

File: Data Structures & Algorithms/jump-game-ii/submission-1.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int res = 0;

        int l = 0, r = 0;
        while (r < nums.size() - 1) {
            int start = l, end = r;

            for (int i = start; i <= end; i++) 
                r = max(r, i + nums[i]);

            l = l + 1;
            res++;
        }

        return res;
    }
};
